File: CSES/C++/Introductory Problems/Apple Division.cpp
#include <iostream>
#include <vector>
#include <climits>
#include <cmath>
using namespace std;

int appleDivision(int n, vector<int>& p) {
    int totalSum = 0;
    for (int i = 0; i < n; i++) {
        totalSum += p[i];
    }
    
    int minDiff = INT_MAX;
    for (int i = 1; i < (1 << n); i++) {
        int currSum = 0;
        for (int j = 0; j < n; j++) {
            if (i & (1 << j)) {
                currSum += p[j];
            }
        }
        minDiff = min(minDiff, abs(totalSum - 2 * currSum));
    }
    
    return minDiff;
}

int main() {
    int n;
    cin >> n;
    vector<int> p(n);
    for (int i = 0; i < n; i++) {
        cin >> p[i];
    }
    
    int result = appleDivision(n, p);
    cout << result << endl;
    return 0;
}

File: CSES/Python/Introductory Problems/Apple Division.py
n = int(input())
p = list(map(int, input().split()))

def apple_division(n, p):
    total_sum = sum(p)
    min_diff = float('inf')
    for i in range(1, 1 << n):
        curr_sum = 0
        for j in range(n):
            if i & (1 << j):
                curr_sum += p[j]
        min_diff = min(min_diff, abs(total_sum - 2 * curr_sum))
    return min_diff

result = apple_division(n, p)
print(result)
